planning/behavior_path_planner/src/scene_module/start_planner/start_planner_module.cpp

```diff
@@ -492,99 +492,109 @@ void StartPlannerModule::planWithPriority(
   } else {
-    RCLCPP_ERROR(
-      getLogger(),
-      "search_priority should be efficient_path or short_back_distance, but %s is given.",
-      search_priority.c_str());
+    RCLCPP_ERROR(getLogger(), "Invalid search_priority: %s", search_priority.c_str());
     throw std::domain_error("[start_planner] invalid search_priority");
   }
+  return order_priority;
+}
 
-  for (const auto & p : order_priority) {
-    if (is_safe_with_pose_planner(p.first, p.second)) {
-      return;
-    }
+bool StartPlannerModule::findPullOutPath(
+  const std::vector<Pose> & start_pose_candidates, const size_t index,
+  const std::shared_ptr<PullOutPlannerBase> & planner, const Pose & refined_start_pose,
+  const Pose & goal_pose)
+{
+  // Ensure the index is within the bounds of the start_pose_candidates vector
+  if (index >= start_pose_candidates.size()) return false;
+
+  const Pose & pull_out_start_pose = start_pose_candidates.at(index);
+  const bool is_driving_forward =
+    tier4_autoware_utils::calcDistance2d(pull_out_start_pose, refined_start_pose) < 0.01;
+
+  planner->setPlannerData(planner_data_);
+  const auto pull_out_path = planner->plan(pull_out_start_pose, goal_pose);
+
+  // If no path is found, return false
+  if (!pull_out_path) {
+    return false;
   }
 
-  // not found safe path
+  // If driving forward, update status with the current path and return true
+  if (is_driving_forward) {
+    updateStatusWithCurrentPath(*pull_out_path, pull_out_start_pose, planner);
+    return true;
+  }
+
+  // If this is the last start pose candidate, return false
+  if (index == start_pose_candidates.size() - 1) return false;
+
+  const Pose & next_pull_out_start_pose = start_pose_candidates.at(index + 1);
+  const auto next_pull_out_path = planner->plan(next_pull_out_start_pose, goal_pose);
+
+  // If no next path is found, return false
+  if (!next_pull_out_path) return false;
+
+  // Update status with the next path and return true
+  updateStatusWithNextPath(*next_pull_out_path, next_pull_out_start_pose, planner);
+  return true;
+}
+
+void StartPlannerModule::updateStatusWithCurrentPath(
+  const behavior_path_planner::PullOutPath & path, const Pose & start_pose,
+  const std::shared_ptr<PullOutPlannerBase> & planner)
+{
+  const std::lock_guard<std::mutex> lock(mutex_);
+  status_.driving_forward = true;
+  status_.found_pull_out_path = true;
+  status_.pull_out_path = path;
+  status_.pull_out_start_pose = start_pose;
+  status_.planner_type = planner->getPlannerType();
+}
+
+void StartPlannerModule::updateStatusWithNextPath(
+  const behavior_path_planner::PullOutPath & path, const Pose & start_pose,
+  const std::shared_ptr<PullOutPlannerBase> & planner)
+{
+  const std::lock_guard<std::mutex> lock(mutex_);
+  status_.driving_forward = false;
+  status_.found_pull_out_path = true;
+  status_.pull_out_path = path;
+  status_.pull_out_start_pose = start_pose;
+  status_.planner_type = planner->getPlannerType();
+}
```

Comment:
nits:
giving planner_type instead of planner is enough

Reply:
I agree! thanks!

Reply:
changed in https://github.com/autowarefoundation/autoware.universe/pull/5393/commits/791145e6f8368693f6ed74cfa19b6fb33a7a7434